eogrow/utils/map.py

```diff
@@ -6,21 +6,35 @@
 import shutil
 import subprocess
 from tempfile import NamedTemporaryFile
-from typing import List, Union
+from typing import List, Literal, Optional, Union
 
 LOGGER = logging.getLogger(__name__)
 
+GDAL_DTYPE_SETTINGS = {
+    "uint8": "-ot Byte",
+    "uint16": "-ot UInt16",
+    "int8": "-ot Byte -co PIXELTYPE=SIGNEDBYTE",
+    "int16": "-ot Int16",
+    "float32": "-ot Float32",
+}
 
-def cogify_inplace(tiff_file: str, blocksize: int = 2048, nodata: Union[None, int, float] = None) -> None:
+
+def cogify_inplace(
+    tiff_file: str,
+    blocksize: int = 2048,
+    nodata: Union[None, int, float] = None,
+    dtype: Optional[Literal["int8", "int16", "uint8", "uint16", "float32"]] = None,
```

Comment:
you can do `Literal[None, "int8", ...]`, less brackets, same in line 46 and 127

Reply:
fixed